### DISCUSSION TITLE
[IDP Brokering] LDAP / AD Identity Provider Support

### DISCUSSION THREAD

**Discussion by fforootd** [zitadel/zitadel#1938] [Ideas]
ZITADEL should be able to use preexisting users from LDAP / AD systems.

## Sync Features

- Just in time
- Scheduled
  - Full
  - Changed (Updated After timestamp)
- Strategy
  - Read Only
  - Writeback

## Policy

- Disable user management completely for org.

## Authentication Features

- LDAP Binding (plain text and based password)
- Kerberos Support
- MFA usage of ZITADEL

## Mapping 

- Basic Mapping Support (Username, eMail, Profile)

## Availability

We should treat LDAP / AD Providers the same way as we do with OIDC.
If the provider is offline we will need to display an error to the user.

## No Goals / Or for a later release

- Extensiv mapping customisation 
- Group to Roles mappings

Additional inputs? @fgerschwiler @livio-a @adlerhurst

**Comment by fforootd** [comment]
FYI @reinhardmaron

**Comment by fforootd** [comment]
A possible pure go kerberos implementation which supports spnego

https://github.com/jcmturner/gokrb5

**Comment by ibotty** [comment]
Is there any news?

**Reply by hifabienne** [reply]
Hi @ibotty, we don't have any news on this discussion. For the moment we don't have any plans to integrate ldap. If we see that a lot of customers are interested in a topic we will have a closer look at it.

**Reply by fforootd** [reply]
There is the possibilty to combine ZITADEL with a OpenID Connect Facade like Dex to make at least LDAP Authentication work.

If you are interested I can try to write a guide for this.

**Reply by mffap** [reply]
@ibotty we are planning to ship LDAP early next year. Please see https://github.com/zitadel/zitadel/issues/4609 for more information.

**Reply by mffap** [reply]
@ibotty LDAP support has been released for ZITADEL

**Comment by adyanth** [comment]
I would be interested in this too. This is the main component keeping me from using this instead of Keycloak

**Reply by fforootd** [reply]
Ok, good to know. I think at one point we really will expand ZITADEL as LDAP Client.

@hifabienne @livio-a I think we should discuss this again internally.

**Reply by mffap** [reply]
@adyanth LDAP login has been released with https://github.com/zitadel/zitadel/releases/tag/v2.23.0

**Comment by fforootd** [comment]
Once more the question for LDAP came up on [Reddit ](https://www.reddit.com/r/selfhosted/comments/wlnkl8/comment/ijvsbcp/?utm_source=share&utm_medium=web2x&context=3)

**Comment by cemremengu** [comment]
LDAP is definitely a must for enterprise deployements. Almost all companies want to manage roles/groups/attributes with active directory.

**Reply by fforootd** [reply]
True, we still see many LDAP / AD Deployments.

**Reply by mffap** [reply]
@cemremengu LDAP support has been released now

**Comment by mffap** [comment]
Lack of LDAP support is also a major blocker for healthcare-providers. A pre-requisite for integrating with them is support for Auth through LDAP.

**Reply by fforootd** [reply]
Yes I think we really need to look into creating a LDAP Client to use it for Authentication.
Also I think we should allow to read and write metadata from LDAP to our mgmt API with Actions.

**Comment by hifabienne** [comment]
Related issue: #4609

**Comment by berendt** [comment]
We would also need to connect to a LDAP. In particular Active Directory. I was a bit surprised that it didn't work at all. You can find an Active Directory or something else that can do LDAP almost everywhere in the enterprises. And the things are always hard set there and cannot be discussed.

**Reply by fforootd** [reply]
Yeah we understand the dilemma there. That's why we started the groundwork for ldap 😝 but we do not yet have an ETA

**Comment by reqa** [comment]
To summarize from comments above:
* LDAP login has been released with https://github.com/zitadel/zitadel/releases/tag/v2.23.0
* https://zitadel.com/docs/guides/integrate/identity-providers/ldap

**Reply by mw-0** [reply]
@reqa Is there a plan to include ldap groups?

**Reply by hifabienne** [reply]
At the moment we only have implemented the ldap login.
So far there are no other plans.
If you want to see a new feature implemented, would be great if you can open up a feature request with your requirements so we can track it. Prioritization is based on the need of our customers and the community.

**Comment by genert** [comment]
Any update regarding Kerberos support?

**Reply by KamistixX** [reply]
Would be curious to see it implemented as well !
